src/dfx/src/commands/ledger/mod.rs

```diff
@@ -95,49 +98,92 @@ fn get_icpts_from_args(
     }
 }
 
-async fn send_and_notify(
+pub async fn transfer(
+    agent: &Agent,
+    memo: Memo,
+    amount: ICPTs,
+    fee: ICPTs,
+    to: AccountIdBlob,
+) -> DfxResult<BlockHeight> {
+    let timestamp_nanos = SystemTime::now()
+        .duration_since(UNIX_EPOCH)
+        .unwrap()
+        .as_nanos() as u64;
+
+    let mut waiter = Delay::builder()
+        .with(Delay::count_timeout(30))
+        .exponential_backoff_capped(
+            std::time::Duration::from_secs(1),
+            2.0,
+            std::time::Duration::from_secs(16),
+        )
+        .build();
+    waiter.start();
+
+    let block_height: BlockHeight = loop {
+        match agent
+            .update(&MAINNET_LEDGER_CANISTER_ID, TRANSFER_METHOD)
+            .with_arg(Encode!(&TransferArgs {
+                memo,
+                amount,
+                fee,
+                from_subaccount: None,
+                to,
+                created_at_time: Some(TimeStamp { timestamp_nanos }),
+            })?)
+            .call_and_wait(waiter_with_timeout(expiry_duration()))
+            .await
+        {
+            Ok(data) => {
+                let result = Decode!(&data, TransferResult)?;
+                match result {
+                    Ok(block_height) => break block_height,
+                    Err(TransferError::TxDuplicate { duplicate_of }) => break duplicate_of,
+                    Err(transfer_err) => bail!(transfer_err),
+                }
+            }
+            Err(agent_err) if !retryable(&agent_err) => {
+                bail!(agent_err);
+            }
+            Err(agent_err) => {
+                eprintln!("Waiting to retry after error: {:?}", &agent_err);
+                if let Err(_waiter_err) = waiter.async_wait().await {
+                    bail!(agent_err);
+                }
+            }
+        }
+    };
+
+    Ok(block_height)
+}
+
+async fn transfer_and_notify(
```

Comment:
In what context is `transfer_and_notify` still being used, now that we have `transfer`?

Reply:
`ledger create-canister` and `ledger top-up` both call `transfer_and_notify`, which in turn calls `transfer`.

`ledger transfer` calls `transfer` directly, since it doesn't need to follow up with notify.